browser/src/control/Parts.js

```diff
@@ -491,24 +471,54 @@ L.Map.include({
 			};
 
 			this.uiManager.showInfoModal(dialogId, _('Show sheets'), ' ', ' ', _('OK'), buttonCallback, true, dialogId + '-response');
-			const modal = document.getElementById(dialogId);
-			modal.insertBefore(container, modal.children[0]);
-
-			JSDialog.enableButtonInModal(dialogId, dialogId + '-response', false);
-
-			var checkboxes = document.querySelectorAll('#show-sheets-modal input[type="checkbox"]');
-			checkboxes.forEach(function(checkbox) {
-				checkbox.addEventListener('change', function() {
-					var anyChecked = false;
-					checkboxes.forEach(function(checkbox) {
-						if (checkbox.checked) {
-							anyChecked = true;
-						}
-					});
-					JSDialog.enableButtonInModal(dialogId, dialogId + '-response', anyChecked);
+			this.showPageModalImpl(dialogId);
+		}
+	},
+
+	showPageModalImpl: function(dialogId) {
+		const modal = document.getElementById(dialogId);
+
+		if (!modal) {
+			setTimeout(() => { this.showPageModalImpl(dialogId) }, 10);
+			return;
+		}
```

Comment:
Is there a possibility that this gets called multiple times before this timeout can be serviced? Is there also a chance that the `dialogId` element could never be created and this would just create an infinite (non-blocking) loop?

Assuming this is now asynchronous because of the newly introduced layouting task, I would suggest that this also uses a layouting task (assuming they're serviced in order) and fails if `dialogId` doesn't exist. It feels risky to add something like this and also makes the code harder to understand and debug if things fail.